ruby/ext/google/protobuf_c/upb.c

```diff
@@ -14106,49 +14128,100 @@ static bool end_duration_base(upb_json_parser *p, const char *ptr) {
   return true;
 }
 
-static void start_timestamp_base(upb_json_parser *p, const char *ptr) {
+static int parse_timestamp_number(upb_json_parser *p) {
+  size_t len;
+  const char *buf;
+  char *end;
+  int val;
+
+  /* atoi() and friends unfortunately do not support specifying the length of
+   * the input string, so we need to force a copy into a NULL-terminated buffer. */
+  multipart_text(p, "\0", 1, false);
+
+  buf = accumulate_getptr(p, &len);
+  val = atoi(buf);
```

Comment:
please use strtol() instead so you can check for errors.

Reply:
strtol cannot handle leading 0. It parses 08 as 0.

Reply:
If our regex is already validating the input, then I'm ok with atoi().